python/ray/autoscaler/_private/util.py

```diff
@@ -98,9 +100,28 @@ def fillout_defaults(config: Dict[str, Any]) -> Dict[str, Any]:
     defaults = _get_default_config(config["provider"])
     defaults.update(config)
     defaults["auth"] = defaults.get("auth", {})
+    try:
+        defaults = _fillout_available_node_types_resources(defaults)
+    except Exception as e:
+        # We don't want to introduce new errors with filling available node
+        # types resources feature.
+        cli_logger.verbose_error("{}", str(e))
```

Comment:
```suggestion
        logger.exception("Failed to autodetect node resources")
```

Please don't use str(e), that will lose the stack trace.

Reply:
@richardliaw